docs/style-guidelines.md

```diff
@@ -1,20 +1,313 @@
+{
+    CalculateDistance();
+    MoveToEnemy();
+}
+```
+* **Expression-bodies**
+  * remove brackets in most cases where it is possible (for loops, `if`-`else`, methods, and properties)
+    * ⚠️exception: always use brackets for **Unity callbacks** 
+  * **Properties** - placed on the same line 
+  * **Methods** - placed on the new line
+```csharp
+public bool IsInitialized => isInitialized;    // Property - body placed on the same line =>
+
+private void Awake()                           // Unity-callback - use brackets even when expression-body is possible  
+{
+    var collider = GetComponent<Collider>;
+}
+
+public void Initialize() =>                    // Methods - body placed on the new line after =>
+    SubrcribeToEvents();
+
+private void TrimAll()
+{
+    foreach (string device in InputDevices)    // Braces removed
+        device.Trim();
+}
+```
+
+### Attribute usages
+* **Class**
+  * Use `[RequireComponent(typeof(MyComponent))]` attribute when you cache components via `GetComponents` on `Awake` and `Start` callbacks
+  * Use `[DisallowMultipleComponent]` attribute when there should be only one `Component` of such type on the `GameObject`
+* **Methods**
+  * Use `[PublicAPI]` attribute if `public method` is exposed to be called from **outside** of the solution 
+  * Use `[UsedImplicitly]` attribute for implicitly called methods, such as calls from Unity animation events, via Unity `GameObject.SendMessage()`, `GameObject.BroadcastMessage()` and similar. 
+  * Use `[Button]` attribute from `EasyButtons` (requires reference in `.asmdef` file) instead of `[ContectMenu]` attribute. Consider using `Mode` parameter of this attribute for `Editor`/`PlayMode`-only methods.
+* **Variables (in Inspector)**
+  * Use `[SerializedField]` for exposing variable in the inspector instead of converting it to public variable 
+  * Use CAPITAL letters for `[Header]` attribute 
+  * Use `[Space]` attribute for better grouping of exposed variables 
+  * Use `[Tooltip]` instead of comments for exposed in inspector variables
+
+### Scripting Symbols
+* try to avoid using Scripting Symbols. Often it can be solved via polymorphism 
+* when using it, try to decorate as less code as possible
+* No tabs when starting Scripting Symbols 
+* No line breaks between Scripting Symbols and its body
+ ```csharp 
+#if UNITY_WEBGL
+    // Here goes WEBGL code
+#elif UNITY_STANDALONE
+    // Here goes Standalone code
+#endif
+```
+### Comments
+* always use XML comments `/// <summary>`  before the `public` class declaration providing appropriate descriptions of their behavior
+ ```csharp 
+/// <summary>
+/// This is the InitialScene entry point.
+/// Most of the application subsystems should be initialized from this class Awake() event.
+/// </summary>
+public class Main: MonoBehaviour
+{
+    /// ...
+}
+```
+* **only** use comments `//` inside of the class when it is necessary and there is a need to provide additional information which cannot be covered by good naming:
```

Comment:
What is the rationale behind this?

Reply:
Just following best practices and avoiding ofuscating smelly code with smelly comments.
https://www.codeguru.com/csharp/commenting-best-practices-in-c/
https://www.dotnetstuffs.com/code-commenting-best-practices-csharp/
https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/
https://www.codingem.com/how-to-comment-code/

Specifically cases like this:
![image](https://user-images.githubusercontent.com/50231608/196643996-a831425a-a9f0-461b-af45-c81bb2e461de.png)

![image](https://user-images.githubusercontent.com/50231608/196644616-552430be-49cd-4069-a707-3e48dd72b365.png)